nilearn/_utils/data_gen.py

```diff
@@ -744,202 +751,625 @@ def basic_confounds(length, random_state=0):
     return confounds
 
 
-def create_fake_bids_dataset(base_dir='',
-                             n_sub=10,
-                             n_ses=2,
-                             tasks=['localizer', 'main'],
-                             n_runs=[1, 3],
-                             with_derivatives=True,
-                             with_confounds=True,
-                             confounds_tag="desc-confounds_timeseries",
-                             no_session=False,
```

Comment:
Note that this PR changes the API of this function.

`no_session` is removed and to create a dataset with no session folder one must pass `n_ses=0`.

As this is in a private module, I believe there is no need to go through a deprecation cycle, correct?

Reply:
Agreed.

Reply:
Actually, the function name should have a leading underscore as it is private.